app.js

```diff
@@ -72,5 +75,29 @@ if (app.get('env') === 'development') {
 // production error handler
 app.use(errorHandlers.productionErrors);
 
+cron.schedule('0 2 * * * *', async () => {
+  try{
+    
+    console.log("starting cron job...");
+    let currentDate =  new Date().toISOString();
+    currentDate = currentDate.substring(0,currentDate.length -1);
+
+    // Update expired Invoice
+    const invoiceUpdated = await Invoice.updateMany(
+        { expiredDate: { $lt:  currentDate }, status: { $ne: 'overdue' }  },
+        { $set: { status: 'overdue' } }
+      );
+  
+    // Update expired quotes
+    const quoteUpdated = await Quote.updateMany(
+        { expiredDate: { $lt:  currentDate }, status: { $ne: 'expired' }  },
+        { $set: { status: 'expired' } }
+    );
+    console.log('cron job compeleted');
+  }catch(e){
+    console.error(e)
+  }
+}, {timezone: "Asia/Kolkata"});
```

Comment:
move this function to middleware/cronJob.js

make it generic , user can pass param when it will run / timezone as param (default is london)
pass callbak func inside cronJob

then use this middelwere in invoiceController & quoteController , adn pass this code as callback func inside it 

`    let currentDate =  new Date().toISOString();
    currentDate = currentDate.substring(0,currentDate.length -1);

    // Update expired Invoice
    const invoiceUpdated = await Invoice.updateMany(
        { expiredDate: { $lt:  currentDate }, status: { $ne: 'overdue' }  },
        { $set: { status: 'overdue' } }
      );

    // Update expired quotes
    const quoteUpdated = await Quote.updateMany(
        { expiredDate: { $lt:  currentDate }, status: { $ne: 'expired' }  },
        { $set: { status: 'expired' } }
    );
    console.log('cron job compeleted');`

Reply:
@salahlalami 
- Created a general function
![image](https://github.com/idurar/erp-crm/assets/100110395/a3d5c909-cee6-4a2d-b5d8-bcc9b4302e16)

- Que) Are you referring to passing the middleware here, I didn't get you. 
![image](https://github.com/idurar/erp-crm/assets/100110395/6f09f2eb-2e9e-410d-9330-acee6688ad39)

- If we are passing the middleware in the separate controllers in the quote and invoice controller then we can split this function as 
```
const invoiceUpdated = await Invoice.updateMany(
    // code
  );
```
and 
```

const quoteUpdated = await Quote.updateMany(
    //code
);
```